datanode/datanode.go

```diff
@@ -0,0 +1,105 @@
+package main
```

Comment:
I would argue all functions in this file need short comments.
Generally, public functions should have comments.